sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_shared/async_challenge_auth_policy.py

```diff
@@ -90,7 +90,5 @@ async def on_challenge(self, request: "PipelineRequest", response: "PipelineResp
 
         return True
 
-    @property
     def _need_new_token(self) -> bool:
-        # pylint:disable=invalid-overridden-method
```

Comment:
Context: the reason for this change is that the base AsyncBearerTokenCredentialPolicy (from `azure-core`) has a `_need_new_token` method instead of property, even though the synchronous policy class has it as a property. Inconsistency aside, this just adopts the same style to better align with the parent class